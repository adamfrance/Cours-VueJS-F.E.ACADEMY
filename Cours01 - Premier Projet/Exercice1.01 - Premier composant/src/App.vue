<template>
  <Exercice />
</template>

<script>
import Exercice from "./components/Exercice1-01.vue";

export default {
  components: {
    Exercice,
  },
};
</script>
