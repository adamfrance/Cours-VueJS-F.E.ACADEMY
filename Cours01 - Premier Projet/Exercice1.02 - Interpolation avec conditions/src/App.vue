<template>
  <Exercice />
</template>

<script>
import Exercice from "./components/Exercice1-02.vue";

export default {
  components: {
    Exercice,
  },
};
</script>
