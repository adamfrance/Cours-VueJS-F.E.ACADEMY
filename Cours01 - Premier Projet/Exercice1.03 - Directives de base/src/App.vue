<template>
  <div>
  <div>Explorer les directives de base (v-text, v-once, v-html, v-bind, v-if, v-show)</div>
  <Exercice />
  </div>
  
</template>

<script>
import Exercice from "./components/Exercice1-03.vue";

export default {
  components: {
    Exercice,
  },
};
</script>
