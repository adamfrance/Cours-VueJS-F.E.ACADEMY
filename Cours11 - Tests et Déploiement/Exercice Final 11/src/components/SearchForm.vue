<template>
	<div>
		<form @submit.prevent="onSubmit()">
			<input v-model="searchTerm" placeholder="Search" type="text"> 
			<button type="submit" id="search">Search</button>
		</form>
	</div>
</template>

<script>
export default {
	data() {
		return {
			searchTerm: this.$route.query.q || ''
		}
	},
	methods: {
		onSubmit() {
			if(this.searchTerm === '') return;
			this.$router.push({ 
				path:'/',
				query: {
					q: this.searchTerm
				}
			});
		}
	}
}
</script>

<style scoped>
div {
	margin-top: 10px;
	margin-bottom: 10px;
}
</style>