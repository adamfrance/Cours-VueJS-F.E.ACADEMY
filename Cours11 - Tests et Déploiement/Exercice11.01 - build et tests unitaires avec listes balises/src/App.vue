<script setup>
import PostList from './components/PostList.vue'
</script>

<template>
  <header>
  <h1>The Vue.js Workshop Blog</h1>
  </header>

  <PostList :posts="posts" />
</template>

<style>
</style>

<script>
export default {
  data() {
    return {
      posts: [
        {
          title: 'Vue.js for React developers',
          description: 'React has massive popularity here are the key benefits of Vue.js over it.',
          tags: ['vue', 'react'],
        },
        {
          title: 'Migrating an AngularJS app to Vue.js',
          description: 'With many breaking changes, AngularJS developers have found it easier to retrain to Vue.js than Angular 2',
          tags: ['vue', 'angularjs']
        }
      ]
    }
  }

}
</script>

