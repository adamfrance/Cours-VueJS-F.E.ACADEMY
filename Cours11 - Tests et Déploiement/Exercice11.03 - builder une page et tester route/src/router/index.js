import { createRouter, createWebHistory } from 'vue-router';
import PostList from '@/components/PostList.vue';
import SinglePost from '@/components/SinglePost.vue';
import TagPage from '@/components/TagPage.vue';

const routes = [
	{
		path: '/',
		component: PostList
	},
	{
		path: '/:postId', 
		component: SinglePost
	},
	{
		path:'/tags/:tagName',
		component: TagPage
	}
];

const router = createRouter({
  history: createWebHistory(), 
  routes
});

export default router;