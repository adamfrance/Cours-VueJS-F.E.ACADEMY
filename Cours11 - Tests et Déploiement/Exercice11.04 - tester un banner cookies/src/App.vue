<script setup>
import NewsletterBanner from './components/NewsletterBanner.vue';
import CookieBanner from './components/CookieBanner.vue';
</script>

<template>

  <NewsletterBanner />
  <header>
  <h1>The Vue.js Workshop Blog</h1>
  </header>

  <router-view :posts="posts"></router-view>

  <CookieBanner />
</template>

<style>
</style>

<script>
export default {
  data() {
    return {
      posts: [
        {
          slug: 'vue-react',
          title: 'Vue.js for React developers',
          description: 'React has massive popularity here are the key benefits of Vue.js over it.',
          content:'This is the content of the Vue.js for React developers post.',
          tags: ['vue', 'react'],
        },
        {
          slug: 'vue-angularjs',
          title: 'Migrating an AngularJS app to Vue.js',
          description: 'With many breaking changes, AngularJS developers have found it easier to retrain to Vue.js than Angular 2',
          content: 'This is the content of the Vue.js for AngularJS developers post.',
          tags: ['vue', 'angularjs']
        }
      ]
    }
  }

}
</script>

