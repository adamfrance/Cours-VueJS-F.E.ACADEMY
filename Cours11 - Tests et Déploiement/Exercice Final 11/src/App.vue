<script setup>
import NewsletterBanner from './components/NewsletterBanner.vue';
import CookieBanner from './components/CookieBanner.vue';
import SearchForm from './components/SearchForm.vue';
import { ref, computed } from 'vue';
import { useRoute } from 'vue-router';

const route = useRoute();

const posts = ref([
      {
        slug: 'vue-react',
        title: 'Vue.js for React developers',
        description: 'React has massive popularity here are the key benefits of Vue.js over it.',
        content:'This is the content of the Vue.js for React developers post.',
        tags: ['vue', 'react'],
      },
      {
        slug: 'vue-angularjs',
        title: 'Migrating an AngularJS app to Vue.js',
        description: 'With many breaking changes, AngularJS developers have found it easier to retrain to Vue.js than Angular 2',
        content: 'This is the content of the Vue.js for AngularJS developers post.',
        tags: ['vue', 'angularjs']
      }
    ]
)


const relevantPosts = computed(() => {
  if(!route) return posts;
  const { q } = route.query;
  if(!q) return posts.value;
  return posts.value.filter(
    p => p.title.toLowerCase().includes(q.toLowerCase())
  )
})
</script>

<template>

  <NewsletterBanner />

  <SearchForm />

  <header>
  <h1>The Vue.js Workshop Blog</h1>
  </header>

  <router-view :posts="relevantPosts"></router-view>

  <CookieBanner />
</template>


