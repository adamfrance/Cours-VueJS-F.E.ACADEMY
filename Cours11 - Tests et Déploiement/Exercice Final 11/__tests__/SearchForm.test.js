import { describe, it, expect, vi } from 'vitest'

import { shallowMount, mount, flushPromises } from '@vue/test-utils';
import SearchForm from '../src/components/PostList.vue';
import App from '../src/App.vue';

import router from '@/router';
import { createTestingPinia } from '@pinia/testing'

describe('SearchForm', () => {

  it('PostList renders tags for each post 2', () => {
    
    const wrapper = shallowMount(SearchForm, {
	  global: {
	  	plugins: [ router ]
	  }
    })

	expect(wrapper.text()).toMatchSnapshot();
  });

  it('SearchForm filter by keyword on submission', async() => {

	router.push('/');
	await router.isReady();

    const wrapper = mount(App, {
	  global: {
	  	plugins: [ router, createTestingPinia({createSpy:vi.fn}) ]
	  }
    })

  	expect(wrapper.text()).toContain('Migrating an AngularJS app to Vue.js');
  	expect(wrapper.text()).toContain('Vue.js for React developers');

	wrapper.find('input[type="text"]').setValue('react');

	await wrapper.find('button#search').trigger('click');
	await wrapper.find('form').trigger('submit');

	await flushPromises();


  	expect(wrapper.text()).not.toContain('Migrating an AngularJS app to Vue.js');
  	expect(wrapper.text()).toContain('Vue.js for React developers');


  });

})
