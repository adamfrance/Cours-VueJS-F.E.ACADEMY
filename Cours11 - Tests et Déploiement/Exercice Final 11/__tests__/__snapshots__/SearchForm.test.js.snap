// Vitest Snapshot v1

exports[`SearchForm > PostList renders tags for each post 2 1`] = `""`;
